### DISCUSSION TITLE
Need help to create my model

### DISCUSSION THREAD

**Discussion by ashokambig** [neperfepx/neper#339] [Q&A] [answered]
Hi,
I am currently working on fracture modelling of a porous rock using Abaqus.
I thought neper is a good tool to model the porous media.
The idea is to create a microstructure using neper. After the mesh is created, I will delete certain  elsets to match the required porosity.
Pore dimensions are in the nonometer range (say 20nm) and therefore it is quite tricky to create the mesh without compromising on the number of elements so that the solution time in Abaqus is limited. 
I gave my first try to mesh a volume of 2cmx2cmx1.5cm with 6000 grains. The resulting mesh has around 5.8 million nodes which is alot
Attached is the neper file I have used. [cube_in_another_cube.txt](https://github.com/rquey/neper/files/7625833/cube_in_another_cube.txt)

My main aim is to get a porous rock model with porosity of 5%. I have written a code in python which will randomly pick the elsets with certain volumes and delete those to generate the final required inp file for abaqus. But currently my model looks so fine. It seems like with this approach I can not get the mesh at the nanometer level. As the Model is expected to be much larger, number of elements/nodes are expected to be way beyond the capacity of an average solver.
As my  focus is only the pores, only the boundaries sorrouding these  pores can be of fine of mesh and rest of the model can be coarsely meshed.
One  way to imagine here is, to think of neper model with only 5% of the model as fine crystals and rest as coarse crystals. Pore distribution can be random.
Is there anyway way we can achieve this in Neper? Any help would be of great help.
Thanks

**Comment by dikshanshs** [comment]
@ashokambig either scale dimension to a smaller size so that amount of grains is lowered or increase rcl, these will result in a decreased number of nodes.

**Comment by rquey** [comment, marked as answer]
You coud define a bimodal cell size distribution (1 for the normal grains, 1 for the pores), define a group for each, and then assign an appropriate element characteristic length to each group:

An example in 2D:

```
neper -T -n from_morpho -morpho "diameq:lognormal(0.1,0.01)+0.5*lognormal(0.01,0.001),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.03?1:2" -morphooptiini "weight:radeq,coo:packing" -dim 2 -o 2d
neper -M 2d.tess -rcl "group==1?1:0.1" -pl 10
```

Visualize:
```
neper --rcfile none -V 2d.tess,2d.msh -dataelt2dedgerad 0.0005 -print img
```
![image](https://user-images.githubusercontent.com/2659376/148394390-5eaadc5d-cb6c-47af-ba97-865c55834054.png)

The element clusters are where the pores are.

In `-T`, `-transform "rmcell(...)"` should make it possible to remove the cells associated to the pores, but the option is buggy at the moment...

**Reply by ashokambig** [reply]
Thank you for your prompt Answer.

Regards
Ashok

**Comment by ashokambig** [comment]
Hi Anuj,

I really can not give you the perfect answer.
But may be you can refer the neper help document on that.
Regarding your question on the porosity, I am able to generate them in the
abaqus input file that is generated by Neper.
So this is the case of porosity in the meshed file. I have used python to
generate required porosity.
The programming in python involves the deletion of the grains randomly
until the required porosity is met.
May be you can do this also in the geometry model using the TESS file I
guess.
You have to find it somehow.
All the best.

Ashok

**Reply by anujyadav0693** [reply]
thank you for your valuable reply.

**Reply by riddraw** [reply]
Hi Ashok, would it be possible for you to share the Python script or provide some guidance on how you achieved this? It would be really helpful for my current project.

I look forward to your response and appreciate your help!

**Comment by ashokambig** [comment]
Hi Anuj,

I really can not give you the perfect answer.
But may be you can refer the neper help document on that.
Regarding your question on the porosity, I am able to generate them in the
abaqus input file that is generated by Neper.
So this is the case of porosity in the meshed file. I have used python to
generate required porosity.
The programming in python involves the deletion of the grains randomly
until the required porosity is met.
May be you can do this also in the geometry model using the TESS file I
guess.
You have to find it somehow.
All the best.

Ashok
